lib/authenticate_methods.rb

```diff
@@ -38,66 +38,41 @@ def use_signed_params
     incoming_user = nil
     found_incoming_user_by = nil
 
-    # First try to to find an existing user by an external UUID
-
+    # Try to to find an existing user who has a matching external UUID
+    # which indicates the account has been linked before
+    # if found, we trust it and sign the user in automatically
     if external_user_uuid.present?
       incoming_user = UserExternalUuid.find_by_uuid(external_user_uuid).try(:user)
-
-      if incoming_user.present?
-        found_incoming_user_by = "external uuid"
-      end
-    end
-
-    # If that didn't pan out, try to find an existing user with a verified email
-    # matching the secure params email
-
-    if incoming_user.nil? && external_email.present?
-      incoming_user = LookupUsers.by_verified_email(external_email).first
-
       if incoming_user.present?
-        found_incoming_user_by = "external email"
-
-        if external_user_uuid.present?
-          # Go ahead and track the external UUID so can use in future
-          incoming_user.external_uuids.create!(uuid: external_user_uuid)
-        end
+        sign_out!(security_log_data: {type: 'new external user'}) if signed_in?
```

Comment:
This is not necessarily a "new external user" because the incoming user could be the same as the currently-logged-in user... so I think we still need the `if signed_in? && incoming_user != current_user` logic and if that triggers we sign out for a `different external user` reason.  (if it is the same user logging in again, we just don't do the `sign_out!` but we do repeat the `sign_in!` so that a security log is entered.)